pkg/util/cert/secret.go

```diff
@@ -39,6 +39,7 @@ func CopySecret(kubeClient kubernetes.Interface, sourceNamespace string, sourceN
 		existingTargetSecret.Name = targetName
 		existingTargetSecret.UID = ""
 		existingTargetSecret.ResourceVersion = ""
+		existingTargetSecret.OwnerReferences = nil
```

Comment:
The owerReferences of the source secret is:
```
 ownerReferences:
    - apiVersion: proxy.open-cluster-management.io/v1alpha1
      kind: ManagedProxyConfiguration
      name: cluster-proxy
      uid: 394880fd-9d6b-485b-a1bd-c3ae8feea023
      blockOwnerDeletion: true
```

But cluster-gateway doesn't(and shouldn't) have this permission.

Reply:
Error log:
```
E1008 07:44:17.090829 1 controller.go:317] controller/clustermanagementaddon "msg"="Reconciler error" "error"="failed to ensure required proxy client related credentials: failed copy secret proxy-server-ca: failed creating CA secret: secrets \"proxy-server-ca\" is forbidden: cannot set blockOwnerDeletion if an ownerReference refers to a resource you can't set finalizers on: , <nil>" "name"="cluster-gateway" "namespace"="" "reconciler group"="addon.open-cluster-management.io" "reconciler kind"="ClusterManagementAddOn"

```